### DISCUSSION TITLE
Migrating Terraform state paths to Terragrunt

### DISCUSSION THREAD

**Discussion by JCapriotti** [gruntwork-io/knowledge-base#753] [How do I]
Hello,
I am investigating Terragrunt by way of converting an existing Terraform structure. Everything is pretty straightforward, except for how to handle a change in the resource names in our remote state file.

For example, each of our existing environment TF folders references a module like this:

```terraform
module "api" {
  source = "../../module"
}
```

So in the existing TF state, all the resource names are prefixed with `module.api`, e.g. `module.api.aws_lb_listener_rule.services`.

When running `terragrunt plan` everything is showing up as a destroy + create because of the module above. What was `module.api.aws_lb_listener_rule.services` in the old Terraform is now `aws_lb_listener_rule.services` in Terragrunt.

I wasn't able to find a quick way or any guidance for how to handle this. To get past it, I put together a bunch of `state mv` commands like:

```shell
terragrunt state mv module.api.aws_lb_listener_rule.services aws_lb_listener_rule.services
```

Is there a better way to do this that I missed?

Thanks,
Jason

---

<ins datetime="2023-07-12T19:13:53Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110340">Tracked in ticket #110340</a></p>
</ins>

**Comment by pras111gw** [comment]
Hi @JCapriotti please see: https://github.com/orgs/gruntwork-io/discussions/229

**Reply by assiakhateeb** [reply]
Hey @pras111gw 
I'm encounter this issue while migrating from Terraform to Terragrunt:
```
  # module.common_helm.helm_release.kube_state_metrics will be destroyed
  # (because helm_release.kube_state_metrics is not in configuration)
 ```
Terragrunt renames `module.common_helm.helm_release.kube_state_metrics` to `helm_release.kube_state_metrics` after destroying it and creating a new one.

I tried your suggestion and ran the following commands from the directory containing `.terragrunt-cache`:
```
terragrunt state pull > state-backup.tfstate 
terragrunt state push state-backup.tfstate
```
However, the issue wasn't solved in the plan output, and Terragrunt still plans to destroy the resource `module.common_helm.helm_release.kube_state_metrics` and create `helm_release.kube_state_metrics`.

Thanks!
